### ISSUE TITLE
Error during local development setup.

### ISSUE THREAD

**Issue by Ari1009** [internetarchive/openlibrary#7725]
Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "docker/ol-solr-updater-start.sh": stat docker/ol-solr-updater-start.sh: no such file or directory: unknown , Error still shows up even when i followed the guide  [line ending,symlinks and git submodules](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#fix-line-endings-symlinks-and-git-submodules-only-for-windows-users-not-using-a-linux-vm).I am using an arcolinux system

### Evidence / Screenshot (if possible)
![Screenshot_2023-03-24-08-50-25_1920x1080](https://user-images.githubusercontent.com/69034460/227581688-c8fe8f81-ab60-4314-b5cd-22ded798eab5.png)

### Steps to Reproduce

1. Go to  openlibrary folder
2. Do ... docker compose up

* Actual:shows file not exist
* Expected:installation success and deployed at 8080

**Comment by scottbarnes** [comment]
@Ari1009, thanks for raising this issue. I realize these are sort of basic questions, but to help us understand what's going on, as this has happened in different contexts, what's the output of:
1. `ls -l docker/ol-home-start.sh`
2. `file docker/ol-home-start.sh`

**Comment by Ari1009** [comment]
![Screenshot_2023-03-24-10-23-17_1920x1080](https://user-images.githubusercontent.com/69034460/227590474-f9dbdb01-1c96-4d81-869a-9c93fc55ed53.png)

These are the output for the above mentioned commands.

**Comment by scottbarnes** [comment]
Well this is mysterious. I am grasping at straws already. I don't see why it would make a difference, but have you tried rebuilding everything with something like `docker compose build --no-cache --pull`?

The files appear to all have `LF` endings, which makes sense, given that you are (1) on Linux and (2) did steps to double check line endings, but just to try to verify every step, I guess it makes sense to also check that `docker-compose.override.yml` is also `LF`.

@cclauss, @cdrini, any ideas?

**Comment by scottbarnes** [comment]
@Ari1009, I put ArcoLinux D into a virtual machine (using QEMU) to see if I could replicate the issue, but `docker compose build` and `docker compose up` went all right for me and I was able to access the local development server.

I am sharing my command line history in the off chance that helps us find new angles to explore:
```sh
yay -S docker docker-compose
sudo systemctl enable docker.service
sudo systemctl start docker.service
docker --version        # Docker version 23.0.1, build a5ee5b1dfc
docker compose version  # Docker Compose version 2.17.0
sudo usermod -aG docker scott
<log out, log in>
ssh-keygen -t ed25519 -C "docker-investigation"
eval "$(ssh-agent -s)"
ssh-add ~/.ssh/id_ed25519
cat .ssh/id_ed25519.pub 
<add ssh key to GitHub>
git clone git@github.com:scottbarnes/openlibrary.git
cd openlibrary/
git remote add upstream git@github.com:internetarchive/openlibrary.git
git pull upstream master
docker compose build
docker compose up
```

At this point the containers came up normally and I was able to access the local development site via a web browser.

**Comment by Ari1009** [comment]
At first, I thought my issue was due to being behind a proxy server. Since Docker Compose showed a failed installation of pip, the commands were not able to be executed in the container. I solved that by downloading from my network data, and you can also add "pip install --proxy". However, when running "docker-compose up", the installation was not present, and I received an error that matched the one described in the guide. This confused me as well as to why it is not working.

**Comment by Ari1009** [comment]
Finally, I got it working. It appears that Docker and Docker Compose were in conflict, so I completely removed Docker and then reinstalled it,This seemed to solve the issue. 
However, I'm now facing another problem: when I try to access localhost:8080, it shows a blank page and does not load the page. But when I halt the process, it shows up, which means it works, but it doesn't load. I'm currently trying to understand this error.
![Screenshot_2023-03-25-11-53-48_1920x1080](https://user-images.githubusercontent.com/69034460/227701356-b467d122-299a-43a4-9048-7f4e7a069e29.png)

**Comment by scottbarnes** [comment]
Ah, good call on trying to reinstall Docker. I am glad that worked.

With respect to the first screenshot, I wonder if archive.org was just simply having some connectivity issues (or possibly the container itself is having connectivity issues).

If you get more errors like this where you see timeouts or retry errors when connecting to external hosts, it might be worth checking connectivity from within the web container with something like:
```
docker compose exec -u root web bash
apt-get install iputils-ping
ping 8.8.8.8  # Or whatever hosts you know to be up and working
```
If `apt-get` can't connect to the apt servers to fetch the package, that would indicate there is a larger Docker/network connectivity issue at play as well.

With respect to the second screenshot, I see the URL bar says `/account/create`; did `/account/login` look like this and could you log in?
![image](https://user-images.githubusercontent.com/26524678/227724207-14c2043d-5746-444c-90db-5bc26b195bc8.png)

**Comment by Ari1009** [comment]
Yup, the actual error was caused by Docker Desktop. It had worked fine before for other apps, So I never realized until now that there is a package clash between Docker Compose and Docker Desktop for Arch Linux. I think it will be better to update the documents to specify to install only Docker and Docker Compose, and remove Docker Desktop for arch linux [(error)](https://github.com/docker/desktop-linux/issues/133)

Regarding the second question, yes, the login page shown is the same. I have made some updates to the sign page [(reference)](https://github.com/internetarchive/openlibrary/pull/7729) and am trying to understand the codebase better. I hope to learn more and contribute to it.
